### PR TITLE
[SPARK-11706][Streaming]Fix the bug that Streaming Python tests cannot report failures

### PR DESCRIPTION
**Pull request by zsxwing**
This PR just checks the test results and returns 1 if the test fails, so that `run-tests.py` can mark it fail.

### CODE REVIEW

**Comment by zsxwing** [comment]
Manually tested this patch locally by adding `1 / 0` to `test_map`. It did report the failure after applied this patch.

File: python/pyspark/streaming/tests.py [hunk 1]
```diff
@@ -1322,11 +1322,16 @@ def search_kinesis_asl_assembly_jar():
             "or 'build/mvn -Pkinesis-asl package' before running this test.")
 
     sys.stderr.write("Running tests: %s \n" % (str(testcases)))
+    failed = False
     for testcase in testcases:
         sys.stderr.write("[Running %s]\n" % (testcase))
         tests = unittest.TestLoader().loadTestsFromTestCase(testcase)
         if xmlrunner:
-            unittest.main(tests, verbosity=3,
```
Comment (zsxwing) [inline python/pyspark/streaming/tests.py]: `unittest.main` will call sys.exit after finishing tests, so we should not use it here.

**Comment by zsxwing** [comment]
No surprises. Exposed a lot of test failures. Fixing them

File: python/pyspark/streaming/tests.py [hunk 2]
```diff
@@ -611,12 +611,15 @@ class CheckpointTests(unittest.TestCase):
     @staticmethod
     def tearDownClass():
         # Clean up in the JVM just in case there has been some issues in Python API
-        jStreamingContextOption = StreamingContext._jvm.SparkContext.getActive()
```
Comment (tdas) [inline python/pyspark/streaming/tests.py]: Why this change?

Comment (zsxwing) [inline python/pyspark/streaming/tests.py]: StreamingContext doens't have a static `_jvm`. It's an instance field.

File: python/pyspark/streaming/tests.py [hunk 3]
```diff
@@ -611,12 +611,15 @@ class CheckpointTests(unittest.TestCase):
...
+        jStreamingContextOption = \
+            SparkContext._jvm.org.apache.spark.streaming.StreamingContext.getActive()
         if jStreamingContextOption.nonEmpty():
             jStreamingContextOption.get().stop()
-        jSparkContextOption = SparkContext._jvm.SparkContext.get()
```
Comment (zsxwing) [inline python/pyspark/streaming/tests.py]: `SparkContext` doesn't provide a method to get the active one. So just removed it.

**Comment by tdas** [comment]
Merging this to master, 1.6 and 1.5. Thanks @zsxwing for catching this.
